src/main/groovy/edu/oregonstate/mist/osuevents/resources/CacheResource.groovy

```diff
@@ -0,0 +1,242 @@
+                eventsDAO.deleteFilterItem(it.key, filterID)
+            }
+
+            filterItemChanges.new.each {
+                eventsDAO.createFilterItem(new CacheObject(
+                        id: it.key,
+                        name: it.value
+                ), filterID)
+            }
+
+            filterItemChanges.updated.each {
+                eventsDAO.updateFilterItem(new CacheObject(
+                        id: it.key,
+                        name: it.value
+                ), filterID)
+
+            }
+            allFilterItemChanges["${filterID}"] = filterItemChanges
+        }
+        allChanges["filterItems"] = allFilterItemChanges
+
+        ok(allChanges).build()
+    }
+
+    @PUT
+    @Path('groups')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateGroups(@Auth AuthenticatedUser _) {
+        def groupsDB = cacheObjectsToMap(eventsDAO.getGroups())
+        def groups = cacheDAO.getGroups()
+        def changes = compareMapPair(groupsDB, groups)
+
+        changes.deleted.each {
+            eventsDAO.deleteGroup(it.key)
+        }
+
+        changes.new.each {
+            eventsDAO.createGroup(new CacheObject(
+                    id: it.key,
+                    name: it.value
+            ))
+        }
+
+        changes.updated.each {
+            eventsDAO.updateGroup(new CacheObject(
+                    id: it.key,
+                    name: it.value
+            ))
+
+        }
+
+        ok(changes).build()
+    }
+
+    @PUT
+    @Path('places')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updatePlaces(@Auth AuthenticatedUser _) {
+        def placesDB = cacheObjectsToMap(eventsDAO.getPlaces())
+        def places = cacheDAO.getPlaces()
+        def changes = compareMapPair(placesDB, places)
+
+        changes.deleted.each {
+            eventsDAO.deletePlace(it.key)
+        }
+
+        changes.new.each {
+            eventsDAO.createPlace(new CacheObject(
+                        id: it.key,
+                        name: it.value
+                ))
+        }
+
+        changes.updated.each {
+            eventsDAO.updatePlace(new CacheObject(
+                        id: it.key,
+                        name: it.value
+                ))
+
+        }
+
+        ok(changes).build()
+    }
+
+    private def cacheObjectsToMap(List<CacheObject> cacheObjects) {
+        def cacheObjectMap = [:]
+
+        cacheObjects.each {
+            cacheObjectMap[new String("${it.id}")] = new String("${it.name}")
+        }
+        cacheObjectMap
+    }
+
+    private def compareMapPair(def oldMap, def newMap) {
+        def newKeys = newMap*.key
+        def oldKeys = oldMap*.key
+
+        def removedKeys = oldKeys - newKeys
+        def addedKeys = newKeys - oldKeys
+        def commonKeys = newKeys - removedKeys - addedKeys
+        def changedKeys = commonKeys.findAll { oldMap[it] != newMap[it] }
+        def unchangedKeys = commonKeys - changedKeys
```

Comment:
Correct. This was also a way to display the code in a more intuitive way since Groovy doesn't have any methods to easily compare two maps using their keys as unique identifiers, then comparing the values.